app/models/observation.rb

```diff
@@ -230,8 +230,8 @@ def when_str
 
   def when_str=(x)
     @when_str = x
-    self.when = x if Date.parse(x)
-    x
+    self.when = x if Date.parse(x) rescue ArgumentError
+    self.when
```

Comment:
Are you sure there's no subtle reason why we used to return the input string here, and not the parsed result?